### ISSUE TITLE
Input to the self.mask_decoder

### ISSUE THREAD

**Issue by tanghaotommy** [amazon-science/self-supervised-amodal-video-object-segmentation#9]
Dear authors, 

Thanks for sharing the codes for this great work! I have a question on the input the mask_decoder in the code:

According to the paper and comments in the code, the input to the mask_decoder should be the embedding and the visible mask. However, [here](https://github.com/amazon-science/self-supervised-amodal-video-object-segmentation/blob/main/models/savos.py#L260) shows the input to the mask_decoder is the embedding and the 5th channel of obj_patches_foward, which seems to be the optical flow x, not the visible mask (from the dataset [here](https://github.com/amazon-science/self-supervised-amodal-video-object-segmentation/blob/main/FishBowl/FishBowl_dataset/dataset.py#L209)). Indexing the visible mask might be obj_patches_foward[..., 3], since obj_patches_foward[..., [0, 1, 2]] is rgb. I am wondering would this make a difference? Or did I miss anything? 

I attach a few plots for different channels of obj_patches_foward
obj_patches_foward[0, 0, ..., 3]:
<img width="541" alt="obj_patches_foward_3" src="https://user-images.githubusercontent.com/15168642/226751994-35dab70f-cdef-4295-8684-074161676636.png">

obj_patches_foward[0, 0, ..., 4]:
<img width="541" alt="obj_patches_foward_4" src="https://user-images.githubusercontent.com/15168642/226752057-cb5e0121-8ef7-4f1e-8bdc-6e8c7a1e11a3.png">

Thank you in advance and looking forward to your reply!

**Comment by nigelyaoj** [comment]
Good catch!  Actually, we had observed the issue before and found that the results from applying a visible mask and optical flow x are similar, based on our experimentation.
